### PR TITLE
{geo}[foss/2023a] ESMPy v8.6.0

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 5 out of 6 (1 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/14e9f6e51286a474ba5a0bf628f5485d for a full test report.

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso
EB_ARGS="ESMF-8.6.0-foss-2023a.eb ESMPy-8.6.0-foss-2023a.eb"

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="ESMF-8.6.0-foss-2023a.eb ESMPy-8.6.0-foss-2023a.eb"

**Comment by verdurin** [comment]
@lexming could you explain these failures and your workaround?

**Comment by lexming** [comment]
@verdurin sure, it's just that `ESMF` needed a rebuild due to https://github.com/easybuilders/easybuild-easyblocks/pull/3368

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/e727b3604b02428753d13cef47d7b6ec for a full test report.

**Comment by boegel** [comment]
@verdurin Did you rebuild `ESMF` first in your last test?

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3368
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/377e09a0196bbbf926ffcecdcdf5832a for a full test report.

**Comment by laraPPr** [comment]
I rebuild ESMF and not sure what went wrong

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen3

**Comment by lexming** [comment]
@laraPPr your failed test was due to broken download link in the tests, it is fixed now

File: easybuild/easyconfigs/e/ESMPy/ESMPy-8.6.0_use-static-version.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+-    # setuptools-git-versioning shouldn't be needed here, but is
+-    # included as a workaround for problems with the build-time
+-    # installation of this package with python 3.10 (given by the
+-    # build-system section above). By including it here, we at least
+-    # ensure that this package will be available for a second or
+-    # subsequent pip install of esmpy.
+-    'setuptools-git-versioning; python_version >= "3.10"',
+ ]
+-dynamic = [ "version" ]
++version = "8.6.0"
```
Comment (smoors) [inline easybuild/easyconfigs/e/ESMPy/ESMPy-8.6.0_use-static-version.patch:29]: let's replace this with a magic string which can be replaced in the easyconfig with a sed cmd

Comment (lexming) [reply easybuild/easyconfigs/e/ESMPy/ESMPy-8.6.0_use-static-version.patch]: sure, done in https://github.com/easybuilders/easybuild-easyconfigs/pull/20984/commits/12432c888375cae7d733a548324f92c7753677f2

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen3

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @lexming!

**Pull request merged**